topics/reference_guide/custom_language_support/documentation.md

```diff
@@ -10,28 +10,105 @@
+and register it as `com.intellij.platform.backend.documentation.psiTargetProvider` extension point to build documentation
+for PSI elements by overriding `PsiDocumentationTargetProvider.documentationTarget()`.
+
+SymbolDocumentationTargetProvider
+: Implement
+[`SymbolDocumentationTargetProvider`](%gh-ic%/platform/lang-impl/src/com/intellij/platform/backend/documentation/SymbolDocumentationTargetProvider.java)
+and register it as `com.intellij.platform.backend.documentation.symbolTargetProvider` extension point to build documentation
+for [](symbols.md) by overriding `SymbolDocumentationTargetProvider.documentationTarget()`.
+
+### `DocumentationTarget`
+
+Each of the implementations above returns instances of
+[`DocumentationTarget`](%gh-ic%/platform/lang-impl/src/com/intellij/platform/backend/documentation/DocumentationTarget.kt).
+The main work is done in `computeDocumentation()` where the documentation is built from the available
+information.
+If plugin developers worked with the now deprecated
+[`DocumentationProvider`](%gh-ic%/platform/analysis-api/src/com/intellij/lang/documentation/DocumentationProvider.java)
+before, then `computeDocumentation()` should do the work that was formerly done in
+`DocumentationProvider.generateDoc()`.
+
+In addition to showing the documentation, the `computeDocumentationHint()` method returns the text to be displayed
+when the user hovers over an element with <shortcut>Ctrl</shortcut>/<shortcut>Cmd</shortcut> pressed or when
+<ui-path>Preferences | Editor | Code Editing | "Show quick documentation on hover"</ui-path> is enabled.
+In the old framework, this method was called `DocumentationProvider.getQuickNavigateInfo()`.
+
+The `createPointer()` method manages instance restoration and ensures access to the entity across different read actions.
+When implementing the `createPointer()` method, it is essential to handle invalidated PSI elements.
+Unlike PSI elements, the `DocumentationTarget` API does not include an `isValid()` method and the returned pointer is expected
+to be `null` if the instance (and all contained objects) cannot be restored.
+Plugin authors can use
+[`KotlinDocumentationTarget.createPointer()`](%gh-ic%/plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/quickDoc/KotlinDocumentationTarget.kt)
+as a reference.
+
+### Examples
+{#documentationtarget-api-examples}
+
+- [`KotlinPsiDocumentationTargetProvider`](%gh-ic%/plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/quickDoc/KotlinPsiDocumentationTargetProvider.kt).
+- [`KotlinDocumentationTarget`](%gh-ic%/plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/quickDoc/KotlinDocumentationTarget.kt)
+- [`PsiElementDocumentationTarget`](%gh-ic%/platform/lang-impl/src/com/intellij/lang/documentation/psi/PsiElementDocumentationTarget.kt)
+
+## DocumentationProvider API
+
+> As of IntelliJ Platform version 2023.1 the `DocumentationProvider` API is deprecated and plugin
+> authors should use the [](#documentationtarget-api).
```

Comment:
```suggestion
> As of IntelliJ Platform version 2023.1, the `DocumentationProvider` API is deprecated, and it should be migrated to the [](#documentationtarget-api).
```

Reply:
I'll change this to "... and plugins should migrate to.." because the "it" in your sentence would refer to the API which doesn't make sense.